### ISSUE TITLE
TPM event log

### ISSUE THREAD

**Issue by frezbo** [siderolabs/talos#7752]
From the AsGO 2023 conference and systemd https://github.com/systemd/systemd/pull/29004 we need to log the TPM events to prove we did extend PCR and also for future work for remote attestation and systemd-pcrlock functionality for Talos to support kexec 

Ref: https://github.com/systemd/systemd/pull/28891

**Comment by dsseng** [comment]
Currently working on this: added logging of PCR extensions into the main system log, currently working on parsing `/sys/kernel/security/tpm0/binary_bios_measurements` to get prior measurements from firmware and bootloader. What should be done to this data then?

**Comment by frezbo** [comment]
> Currently working on this: added logging of PCR extensions into the main system log, currently working on parsing `/sys/kernel/security/tpm0/binary_bios_measurements` to get prior measurements from firmware and bootloader. What should be done to this data then?

We were holding onto this until we have selinux support, the long term plan is to do something similar to systemd-pcrlock to support disk unlock when firmware updates might change PCR 7 data, also maybe tie to more PCR values for disk encryption

**Comment by frezbo** [comment]
> Currently working on this: added logging of PCR extensions into the main system log

This needs to be a separate log file (hence waiting on landing selinux so only machined can touch the file),  but i guess we can accept the PR

**Comment by dsseng** [comment]
>  to support disk unlock when firmware updates might change PCR 7 data, also maybe tie to more PCR values for disk encryption

Is the goal like https://github.com/okirch/pcr-oracle? Unsure how would that work with firmware update. My openSUSE systems required LUKS key to boot after a firmware update. Maybe we could predict the FW PCR value if the image to be loaded is in a raw format or has PCR values listed.

**Comment by frezbo** [comment]
> > to support disk unlock when firmware updates might change PCR 7 data, also maybe tie to more PCR values for disk encryption
> 
> Is the goal like https://github.com/okirch/pcr-oracle? Unsure how would that work with firmware update. My openSUSE systems required LUKS key to boot after a firmware update. Maybe we could predict the FW PCR value if the image to be loaded is in a raw format or has PCR values listed.

yeh, kind of similar, also this would really help once it lands https://github.com/fwupd/fwupd/issues/6318

**Comment by dsseng** [comment]
Well, should be workable. Not sure how would that work if new firmware, for example, changes the way it hashes NVRAM records or something else like Secure Boot data, but that shouldn't change at least on minor bugfix updates

**Comment by frezbo** [comment]
https://github.com/canonical/tcglog-parser this might be interesting
